examples/text_to_image/train_text_to_image.py

```diff
@@ -112,6 +113,18 @@ def log_validation(vae, text_encoder, tokenizer, unet, args, accelerator, weight
     torch.cuda.empty_cache()
 
 
+@dataclasses.dataclass
+class TrainingState:
+    global_step: int = 0
+
+    def load_state_dict(self, state_dict):
+        for k, v in state_dict.items():
+            setattr(self, k, v)
+
+    def state_dict(self):
+        return dataclasses.asdict(self)
```

Comment:
This class can hold any misc training states we want to serialize. We have to add this for the global step which now is no longer stored as part of the filename